### PR TITLE
The cards expired date validation control bug fixed.

### PR DESCRIPTION
**Pull request by ocakliemre**
The expiry date on the cards should be based on the last day of that month. Since the current structure is based only on the month, it does validation not work correctly. As a bug fix, the last day of the month was found and added to the card expiry date.

### CODE REVIEW

**Comment by DhavalRKansara** [comment]
@ocakliemre This branch is having conflicts with the latest release, can you please take upstream of the latest changes, fix the conflict and update the PR so we can review it?

**Comment by kavantrivedi** [comment]
@ocakliemre  just a reminder for resolving conflicts of this branch with the latest release, can you please update this PR so we can review it?

File: lib/credit_card_form.dart [hunk 1]
```diff
@@ -247,7 +248,8 @@ class _CreditCardFormState extends State<CreditCardForm> {
                           final List<String> date = value.split(RegExp(r'/'));
                           final int month = int.parse(date.first);
                           final int year = int.parse('20${date.last}');
-                          final DateTime cardDate = DateTime(year, month);
+                          final int lastDayofMonth = DateTime(year, Jiffy( DateTime(year,month)).add(months: 1).month, 0 ).day;
```
Comment (kavantrivedi) [inline lib/credit_card_form.dart:251]: @ocakliemre can we use another way to get the last date of the month, as we do not want any additional package dependencies?

**Review by kavantrivedi — APPROVED** [review]

File: lib/credit_card_form.dart [hunk 2]
```diff
@@ -247,7 +245,8 @@ class _CreditCardFormState extends State<CreditCardForm> {
                           final List<String> date = value.split(RegExp(r'/'));
                           final int month = int.parse(date.first);
                           final int year = int.parse('20${date.last}');
-                          final DateTime cardDate = DateTime(year, month);
+                          final int lastDayOfMonth =  month < 12 ? DateTime(year, month + 1, 0).day : DateTime(year + 1, 1, 0).day;
+                          final DateTime cardDate = DateTime(year, month, lastDayOfMonth,23,59,59,999);
```
Comment (shwetachauhan-simform) [inline lib/credit_card_form.dart:249]: Can you please format this code?

File: example/lib/main.dart [hunk 3]
```diff
@@ -190,8 +190,7 @@ class MySampleState extends State<MySample> {
                           style: ElevatedButton.styleFrom(
                             shape: RoundedRectangleBorder(
                               borderRadius: BorderRadius.circular(8.0),
-                            ),
-                            primary: const Color(0xff1b447b),
+                            ), backgroundColor: const Color(0xff1b447b),
```
Comment (shwetachauhan-simform) [inline example/lib/main.dart:193]: same here, formatting issue

**Review by shwetachauhan-simform — CHANGES_REQUESTED** [review]

**Comment by kavantrivedi** [comment]
Hey @ocakliemre,
Can squash these commits, it would be better to get all changes in a single commit.

**Comment by shwetachauhan-simform** [comment]
@ocakliemre : You can squash all commit

**Review by ShwetaChauhan18 — APPROVED** [review]

**Review by ShwetaChauhan18 — APPROVED** [review]

**Review by shwetachauhan-simform — APPROVED** [review]

**Review by shwetachauhan-simform — APPROVED** [review]

**Pull request merged**